From 0e5c6fbd4b6368e88e9f4ca6568affe70963035e Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:57:48 +0200
Subject: [PATCH 28/50] AnnaThermostat: add domestic_hot_water_setpoint
 measurement

Map domestic_hot_water_setpoint to a Domoticz setpoint device in degrees Celsius.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 1121a5bc1..8dae0f672 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1344,6 +1344,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendTempSensor(applianceIDhex + 5, batterypercentage, temperature, ApplianceName + " " + sname);;
                 }
             }
+			else if (sname == "domestic_hot_water_setpoint")
+			{
+				if (!tmpstr.empty())
+				{
+					float setpoint = (float)atof(tmpstr.c_str());
+						SendSetPointSensor(0, 0, (uint8_t)((applianceIDhex >> 8) & 0xFF), (uint8_t)(applianceIDhex & 0xFF), 2, batterypercentage, setpoint, ApplianceName + " " + sname);
+				}
+			}
             else if (sname == "humidity")
             {
                 if (!tmpstr.empty())
-- 
2.51.0

